File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/TokenTextPointer.java
package com.cuupa.classificator.services.kb.semantic.token;

import org.jetbrains.annotations.NotNull;

import java.util.Arrays;

public class TokenTextPointer {

	@NotNull
	private final char[] charArray;
	private final int index;

	public TokenTextPointer(@NotNull char[] charArray, int index) {
		this.charArray = Arrays.copyOf(charArray, charArray.length);
		this.index = index;
	}

	public int getIndex() {
		return index;
	}

	public char get(int i) {
		return charArray[i];
	}

	public int getCharSize() {
		return charArray.length;
	}
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/TokenTextPointer.kt
package com.cuupa.classificator.services.kb.semantic.token

class TokenTextPointer(charArray: CharArray, val index: Int) {

    private val charArrayIntern: CharArray = charArray.copyOf(charArray.size)

    operator fun get(i: Int): Char {
        return charArrayIntern[i]
    }

    val charSize: Int
        get() = charArrayIntern.size

}